packages/sanity/src/core/store/_legacy/document/resolveTypeForDocument.ts

```diff
@@ -14,9 +14,15 @@ export function resolveTypeForDocument(
     return of(specifiedType)
   }
 
-  const query = '*[_id in [$documentId, $draftId]]._type'
-  const documentId = getPublishedId(id)
-  const draftId = getDraftId(documentId)
+  const query = '*[sanity::versionOf($publishedId)]._type'
```

Comment:
```suggestion
  const query = '*[sanity::versionOf($publishedId)]._type[0]'
```
to save us a few transferred bytes and also the map below